src/hooks/useArNSContract/useArNSContract.tsx

```diff
@@ -0,0 +1,45 @@
+import { LocalFileSystemDataProvider } from '../../services/arweave/LocalFilesystemDataProvider';
+import { useStateValue } from '../../state/state';
+import { useState, useEffect } from 'react';
+
+const ARNS_SOURCE_CONTRACT_ID = 'bLAgYxAdX2Ry-nt6aH2ixgvJXbpsEYm28NgJgyqfs-U';
+
+export default function useArNSContract() {
+  const [{ arnsSourceContract }, dispatch] = useStateValue();
+  const [sendingContractState, setSendingContractState] = useState(false);
+
+  useEffect(() => {
+    dispatchNewContractState();
+  }, []);
+  async function dispatchNewContractState(): Promise<void> {
```

Comment:
another nit: add space between these 
```suggestion

  async function dispatchNewContractState(): Promise<void> {
```